packages/grid/src/GridMetricCalculator.ts

```diff
@@ -1423,7 +1423,9 @@ export class GridMetricCalculator {
     userSizes: ModelSizeMap,
     calculateSize: () => number
   ): number {
-    return userSizes.get(modelIndex) ?? calculateSize();
+    // Always re-calculate the size of the item so the calculated size maps are populated
+    const calculatedSize = calculateSize();
+    return userSizes.get(modelIndex) ?? calculatedSize;
```

Comment:
It needs to be calculated in the `getMetrics` anyway - since the text within a cell could change. Refactoring I think I'd have a method to recalculate sizes, which would trigger whenever data is changed.